sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ContainerCreateDeleteWithSameNameTest.java

```diff
@@ -0,0 +1,347 @@
+                    .setLeaseAcquireInterval(Duration.ofSeconds(10))
+                    .setLeaseExpirationInterval(Duration.ofSeconds(30))
+                    .setFeedPollDelay(Duration.ofSeconds(2))
+                    .setLeasePrefix("TEST")
+                    .setMaxItemCount(10)
+                    .setStartFromBeginning(true)
+                    .setMaxScaleCount(0) // unlimited
+                )
+                .buildChangeFeedProcessor();
+
+            try {
+                changeFeedProcessor.start().subscribeOn(Schedulers.elastic())
+                    .timeout(Duration.ofMillis(2 * CHANGE_FEED_PROCESSOR_TIMEOUT))
+                    .subscribe();
+            } catch (Exception ex) {
+                throw ex;
+            }
+
+            // Wait for the feed processor to receive and process the documents.
+            try {
+                Thread.sleep(2 * CHANGE_FEED_PROCESSOR_TIMEOUT);
+                assertThat(changeFeedProcessor.isStarted()).as("Change Feed Processor instance is running").isTrue();
+
+                changeFeedProcessor.stop().subscribeOn(Schedulers.elastic()).timeout(Duration.ofMillis(CHANGE_FEED_PROCESSOR_TIMEOUT)).subscribe();
+
+                for (TestObject item : createdDocuments) {
+                    assertThat(receivedDocuments.containsKey(item.getId())).as("Document with getId: " + item.getId()).isTrue();
+                }
+
+                // Wait for the feed processor to shutdown.
+                Thread.sleep(CHANGE_FEED_PROCESSOR_TIMEOUT);
+
+            } catch (InterruptedException e) {
+                throw new RuntimeException("Interrupted exception", e);
+            }
+        };
+
+        changeFeedCreateDeleteContainerWithSameName(func);
+    }
+
+    @BeforeClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void before_ContainerCreateDeleteWithSameNameTest() throws Exception {
+        client = getClientBuilder().buildAsyncClient();
+        createdDatabase = getSharedCosmosDatabase(client);
+    }
+
+    @AfterClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void after_ContainerCreateDeleteWithSameNameTest() throws Exception {
+        safeDeleteAllCollections(createdDatabase);
+    }
+
+    private <T> void createDeleteContainerWithSameName(Consumer<CosmosAsyncContainer> validateFunc) throws InterruptedException {
+        CosmosAsyncContainer container = null;
+        try {
+            // step1: create container
+            String testContainerId = UUID.randomUUID().toString();
+            CosmosContainerProperties containerProperties = getCollectionDefinition(testContainerId);
+            container = createCollection(this.createdDatabase, containerProperties, new CosmosContainerRequestOptions());
+
+            // Step2: execute func
+            validateFunc.accept(container);
+
+            // step3: delete the container
+            safeDeleteCollection(container);
+            Thread.sleep(2000);
```

Comment:
it might be worth having this as a constant with a quick explanation for why it is needed (emulator's deletes are not instantaneous).